controllers/remote/cluster_cache.go

```diff
@@ -154,8 +159,15 @@ func (t *ClusterCacheTracker) newClusterAccessor(ctx context.Context, cluster cl
 		cancelFunc: cacheCtxCancel,
 	}
 
+	if err := cache.IndexField(ctx, &corev1.Node{}, NodeProviderIDIndex, indexNodeByProviderID); err != nil {
+		return nil, fmt.Errorf("failed to index field %s: %w", NodeProviderIDIndex, err)
+	}
+
 	// Start the cache!!!
 	go cache.Start(cacheCtx)
+	if !cache.WaitForCacheSync(ctx) {
+		return nil, fmt.Errorf("failed waiting for cache for remote cluster %v to sync: %w", cluster, err)
+	}
```

Comment:
This is a good improvement, although could we split it out in a different PR to have its own release note? It might also be worth to backport.

Reply:
I think we need it for the test to not go flaky. How about having its own commit on this PR?